presto-spark-base/src/test/java/com/facebook/presto/spark/TestPrestoSparkAbstractTestQueries.java

```diff
@@ -148,4 +176,71 @@ public void testExplainDdl()
     {
         // DDL statements are not supported by Presto on Spark
     }
+
+    @Test
+    public void testAssignSourceDistributionSplits()
+    {
+        int maxSplitsSize = 2048;
+        Session session = Session.builder(getSession())
+                .setSystemProperty(MAX_SPLITS_DATA_SIZE_PER_SPARK_PARTITION, maxSplitsSize + "B")
+                .build();
+
+        List<ScheduledSplit> splits = new ArrayList<>();
+        long totalSizeInBytes = 0;
+
+        for (int i = 0; i < 1000; ++i) {
+            long splitSizeInBytes = ThreadLocalRandom.current().nextInt((int) (maxSplitsSize * 1.1));
```

Comment:
It doesn't feel like this algorithm is worth to be tested with a fuzz testing. Instead i would recommend adding a number of deterministic test cases to verify all possible corner cases of the algorithm.